lib/core/connection/pool.js

```diff
@@ -643,10 +642,17 @@ function destroy(self, connections, options, callback) {
       return;
     }
 
-    // Zero out all connections
+    // clear all pool state
     self.inUseConnections = [];
     self.availableConnections = [];
     self.connectingConnections = 0;
+    self.executing = false;
```

Comment:
I think we could make a:

```js
Pool.prototype._resetState = function() {
  self.inUseConnections = [];
  self.availableConnections = [];
  self.connectingConnections = 0;
  self.executing = false;
  self.reconnectConnection = null;
  self.numberOfConsecutiveTimeouts = 0;
  self.connectionIndex = 0;
  self.retriesLeft = self.options.reconnectTries;
  self.reconnectId = null;
}
```

and then have destroy call:

```js
self._resetState();
self.queue = [];
```

and have pool.reset just do `this._resetState();`

To elaborate further, could we structure it such that `destroy` just calls `pool.reset()` and then does more?

Reply:
I refactored as much as I think I'm comfortable with here. I'd prefer not to spend too too much time making this connection pool better, and would rather focus on getting your rewrite in.